python/llm/example/Text-Generation-WebUI/modules/callbacks.py

```diff
@@ -41,6 +41,19 @@ def __call__(self, input_ids: torch.LongTensor, _scores: torch.FloatTensor) -> b
         return shared.stop_everything
 
 
+class StopWordsCriteria(transformers.StoppingCriteria):
+    """Custom `StoppingCriteria` which checks if all generated functions in the batch are completed."""
+    def __init__(self, stop_words, tokenizer):
+        self.stop_words = stop_words
+        self.tokenizer = tokenizer
+
+    def __call__(self, input_ids, scores, **kwargs):
+        """Returns true if all generated sequences contain any of the end-of-function strings."""
+        texts =  [self.tokenizer.decode(input_ids[-1][-1])]
+        dones = [any(stop_word in texts for stop_word in self.stop_words)]
```

Comment:
text = self.tokenizer.decode(input_ids[-1][-1])
return text in self.stop_words
??